x-pack/test/functional/apps/canvas/smoke_test.js

```diff
@@ -30,6 +30,10 @@ export default function canvasSmokeTest({ getService, getPageObjects }) {
       await PageObjects.common.navigateToApp('canvas');
     });
 
+    after(async () => {
+      await PageObjects.security.logout();
```

Comment:
It doesn't feel like we should need to logout at the end of this test, this feels more like preparatory work for the canvas feature controls tests... 

I'm sure there's a lot of inconsistency for when which tests logout before/after their run, but I'm starting to think that we should be assuming that when a test suite starts it can be in two states: logged out entirely, logged in as the `elastic` user. A lot of our functional testing services will automatically log a user in using the `elastic` user if they are not already. Likewise, if during testing we log in as a "custom user" we should probably be logging out in the 'after' of the test suite. 

I'm afraid we've gotten away with being rather inconsistent with this approach because we're been writing the functional tests all as part of the "security suite" before some tests that log in as a customer user and then logout in the after.

Reply:
> I'm sure there's a lot of inconsistency for when which tests logout before/after their run, but I'm starting to think that we should be assuming that when a test suite starts it can be in two states: logged out entirely, logged in as the elastic user

Are you suggesting that we should attempt `await PageObjects.security.logout()` before logging in if a test requires a specific user?

> Likewise, if during testing we log in as a "custom user" we should probably be logging out in the 'after' of the test suite.

agreed!